qt/src/SourceManager.cc

```diff
@@ -44,6 +54,32 @@ QDataStream& operator>>(QDataStream& ds, Source*&) {
 	return ds;
 }
 
+Source::Source(const QString& _path, const QString& _displayname, bool _isTemp)
+				: path(_path), displayname(_displayname), isTemp(_isTemp) {
+	// Check whether input PDF file is encrypted
+	if(path.endsWith(".pdf", Qt::CaseInsensitive)) {
+		std::unique_ptr<Poppler::Document> document(Poppler::Document::load(path));
+		if(document && document->isLocked()) {
+			bool ok = false;
+			QString text;
+			while(true) {
+				text = QInputDialog::getText(MAIN, _("Enter password for the PDF file"),
+											 _("Password:"), QLineEdit::Password,
+											 "", &ok);
```

Comment:
Testing it I noticed that, since the dialog is by default very small, the the user most likely won't see the entire window title, but just the "Password:" text. I'd personally use "Enter password for the PDF file" as label and something simple like "Protected PDF" as title.

Reply:
Fixed.